src/geometry/algebra/lie_group.lean

```diff
@@ -0,0 +1,258 @@
+[topological_group G] [lie_group I G]
+{E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+{H' : Type*} [topological_space H'] {I' : model_with_corners 𝕜 E' H'}
+{M : Type*} [topological_space M] [charted_space H' M] [smooth_manifold_with_corners I' M]
+{E'' : Type*} [normed_group E''] [normed_space 𝕜 E'']
+{H'' : Type*} [topological_space H''] {I'' : model_with_corners 𝕜 E'' H''}
+{M' : Type*} [topological_space M'] [charted_space H'' M'] [smooth_manifold_with_corners I'' M']
+
+@[to_additive]
+lemma smooth_mul : smooth (I.prod I) I (λ p : G×G, p.1 * p.2) :=
+lie_group.smooth_mul
+
+@[to_additive]
+lemma smooth.mul {f : M → G} {g : M → G} (hf : smooth I' I f) (hg : smooth I' I g) :
+  smooth I' I (f * g) :=
+smooth_mul.comp (hf.prod_mk hg)
+
+namespace lie_group
+
+/-- `L g` denotes left multiplication by `g` -/
+@[nolint unused_arguments, to_additive L_add "`L_add g` denotes left addition by `g`"]
+def L : G → G → G := λ g : G, λ x : G, g * x
+
+attribute [nolint unused_arguments] lie_add_group.L_add
+
+/-- `R g` denotes right multiplication by `g` -/
+@[nolint unused_arguments, to_additive R_add "`R_add g` denotes right addition by `g`"]
+def R : G → G → G := λ g : G, λ x : G, x * g
+
+attribute [nolint unused_arguments] lie_add_group.R_add
+
+end lie_group
+
+@[to_additive]
+lemma smooth_mul_left {a : G} : smooth I I (λ b : G, a * b) :=
+smooth_mul.comp (smooth_const.prod_mk smooth_id)
+
+@[to_additive]
+lemma smooth_mul_right {a : G} : smooth I I (λ b : G, b * a) :=
+smooth_mul.comp (smooth_id.prod_mk smooth_const)
+
+@[to_additive]
+lemma smooth_on.mul {f : M → G} {g : M → G} {s : set M}
+  (hf : smooth_on I' I f s) (hg : smooth_on I' I g s) :
+  smooth_on I' I (f * g) s :=
+(smooth_mul.comp_smooth_on (hf.prod_mk hg) : _)
+
+lemma smooth_pow : ∀ n : ℕ, smooth I I (λ a : G, a ^ n)
+| 0 := by { simp only [pow_zero], exact smooth_const }
+| (k+1) := show smooth I I (λ (a : G), a * a ^ k), from smooth_id.mul (smooth_pow _)
+
+@[to_additive]
+lemma smooth_inv : smooth I I (λ x : G, x⁻¹) :=
+lie_group.smooth_inv
+
+@[to_additive]
+lemma smooth.inv {f : M → G}
+  (hf : smooth I' I f) : smooth I' I (λx, (f x)⁻¹) :=
+smooth_inv.comp hf
+
+@[to_additive]
+lemma smooth_on.inv {f : M → G} {s : set M}
+  (hf : smooth_on I' I f s) : smooth_on I' I (λx, (f x)⁻¹) s :=
+smooth_inv.comp_smooth_on hf
+
+end lie_group
+
+section prod_lie_group
+
+/- Instance of product group -/
+@[to_additive lie_add_group]
+instance {𝕜 : Type*} [nondiscrete_normed_field 𝕜] {H : Type*} [topological_space H]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]  {I : model_with_corners 𝕜 E H}
+{G : Type*} [topological_space G] [charted_space H G] [group G] [topological_group G]
+[h : lie_group I G] {E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+{H' : Type*} [topological_space H'] {I' : model_with_corners 𝕜 E' H'}
+{G' : Type*} [topological_space G'] [charted_space H' G']
+[group G'] [topological_group G'] [h' : lie_group I' G'] : lie_group (I.prod I') (G×G') :=
```

Comment:
```suggestion
  {E : Type*} [normed_group E] [normed_space 𝕜 E]  {I : model_with_corners 𝕜 E H}
  {G : Type*} [topological_space G] [charted_space H G] [group G] [topological_group G]
  [h : lie_group I G] {E' : Type*} [normed_group E'] [normed_space 𝕜 E']
  {H' : Type*} [topological_space H'] {I' : model_with_corners 𝕜 E' H'}
  {G' : Type*} [topological_space G'] [charted_space H' G']
  [group G'] [topological_group G'] [h' : lie_group I' G'] : lie_group (I.prod I') (G×G') :=
```
Everything in a statement (before the proof) should always be indented by two characters.